nav2_docking/opennav_docking/src/simple_non_charging_dock.cpp

```diff
@@ -268,6 +274,11 @@ void SimpleNonChargingDock::jointStateCallback(const sensor_msgs::msg::JointStat
   is_stalled_ = (velocity < stall_velocity_threshold_) && (effort > stall_effort_threshold_);
 }
 
+opennav_docking_core::DockDirection SimpleNonChargingDock::getDockDirection()
+{
+  return utils::getDockDirectionFromString(dock_direction_);
```

Comment:
Why not set `dock_direction_` as a `DockDirection` type and return it here instead? That way we don't repeatedly need to call `getDockDirectionFromString` and can check the validity of the string on initialization in `configure` so that if its invalid, we can throw an exception or fail configuring?

Ditto for the charging plugin